File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
       Stack<Character> parentheses = new Stack<>();
        for(int i=0; i<s.length(); i++){
            char c = s.charAt(i);
            if(c=='(' || c=='{' || c=='['){
                parentheses.push(c);
            }else {
                if(parentheses.isEmpty()){
                    return false;
                }

                char bracket = parentheses.pop();
                if(bracket=='(' && c!=')'){
                    return false;
                }else if(bracket=='{' && c!='}'){
                    return false;
                } else if(bracket=='[' && c!=']'){
                    return false;
                }

            }
        }
        return parentheses.isEmpty();
    }
}

File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, str: str) -> bool:
        stack = []
        for s in str:
            if s == '(' or s == '[' or s == '{':
                stack.append(s)
            else:
                if not stack:
                    return False
                if s == ')':
                    if stack.pop() != '(':
                        return False
                elif s == ']':
                    if stack.pop() != '[':
                        return False
                elif s == '}':
                    if stack.pop() != '{':
                        return False
        return len(stack)==0

          